crates/subspace-node/src/domain/domain_instance_starter.rs

```diff
@@ -0,0 +1,283 @@
+                        >,
+                    >::new(Some(evm_base_path), domain_cli.additional_args());
+
+                let domain_params = domain_service::DomainParams {
+                    domain_id,
+                    domain_config,
+                    consensus_client,
+                    consensus_network_sync_oracle: consensus_sync_service.clone(),
+                    select_chain,
+                    operator_streams,
+                    gossip_message_sink: xdm_gossip_worker_builder.gossip_msg_sink(),
+                    provider: eth_provider,
+                };
+
+                let mut domain_node = domain_service::new_full::<
+                    _,
+                    _,
+                    _,
+                    _,
+                    _,
+                    _,
+                    evm_domain_runtime::RuntimeApi,
+                    EVMDomainExecutorDispatch,
+                    AccountId20,
+                    _,
+                >(domain_params)
+                .await?;
+
+                xdm_gossip_worker_builder
+                    .push_domain_tx_pool_sink(domain_cli.domain_id, domain_node.tx_pool_sink);
+
+                let cross_domain_message_gossip_worker = xdm_gossip_worker_builder
+                    .build::<CBlock, _, _>(consensus_network_service, consensus_sync_service);
+
+                domain_node
+                    .task_manager
+                    .spawn_essential_handle()
+                    .spawn_essential_blocking(
+                        "cross-domain-gossip-message-worker",
+                        None,
+                        Box::pin(cross_domain_message_gossip_worker.run()),
+                    );
+
+                domain_node.network_starter.start_network();
+
+                domain_node.task_manager.future().await?;
+
+                Ok(())
+            }
+        }
+    }
+}
+
+/// Default sub directory to store network config.
+pub(crate) const DEFAULT_NETWORK_CONFIG_PATH: &str = "network";
+
+/// Create a Configuration object from the current object, port from `sc_cli::create_configuration`
+/// and changed to take `chain_spec` as argument instead of construct one internally.
+fn create_configuration<
+    DCV: DefaultConfigurationValues,
+    CC: CliConfiguration<DCV>,
+    Cli: SubstrateCli,
+>(
+    cli_config: &CC,
+    chain_spec: Box<dyn ChainSpec>,
+    tokio_handle: tokio::runtime::Handle,
+) -> sc_cli::Result<Configuration> {
+    let is_dev = cli_config.is_dev()?;
+    let base_path = cli_config
+        .base_path()?
+        .unwrap_or_else(|| BasePath::from_project("", "", &Cli::executable_name()));
+    let config_dir = base_path.config_dir(chain_spec.id());
+    let net_config_dir = config_dir.join(DEFAULT_NETWORK_CONFIG_PATH);
+    let client_id = Cli::client_id();
+    let database_cache_size = cli_config.database_cache_size()?.unwrap_or(1024);
+    let database = cli_config.database()?.unwrap_or(
+        #[cfg(feature = "rocksdb")]
```

Comment:
we moved away from Rocksdb? any reason why we are still expecting this feature?

Reply:
This function is port from upstream with slight changes related to the chain spec.